src/components/presentational/ViewHeader/ViewHeader.tsx

```diff
@@ -1,13 +1,22 @@
-import React, { useContext } from "react";
-import "./ViewHeader.css"
-import { ColorDefinition, resolveColor } from "../../../helpers/colors";
-import { LayoutContext } from "../Layout";
+import React, { useContext } from 'react';
+import './ViewHeader.css'
+import { ColorDefinition, resolveColor } from '../../../helpers/colors';
+import { LayoutContext } from '../Layout';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
+
+export interface ViewHeaderAction {
+    icon: IconDefinition;
```

Comment:
I would just have this be a React component and not force a font awesome icon...in all places now i'm just using ReactNode so we have flexibility to change it out with something else in the future

Reply:
(We can still have default styling apply when it is a font awesome icon though)

Reply:
Makes sense.  I converted the `action` to a `ReactNode`.